src/features/mangaViewer/components/TopHeader.tsx

```diff
@@ -17,6 +24,8 @@ const TopHeader = () => {
         </div>
         <TopHeaderInfo />
       </div>
+      <ScanModal mangaInfo={mangaInfo} />
+      <LanguageModal mangaInfo={mangaInfo} />
```

Comment:
So as mentioned earlier you can maybe have some boolean that said if the LanguageModal is open or closed and short circuit it using that boolean. `{boolean && mangaInfo && <LanguageModal mangaInfo={mangaInfo} />}`. Not 100% sure though. Let me know

Reply:
The DOM's Dialog modal works best because it automates the pop up feature, including disabling everything in the background.I managed to use a useRef for this.